libminifi/include/io/CRCStream.h

```diff
@@ -224,7 +222,7 @@ template<typename T>
 int CRCStream<T>::readData(uint8_t *buf, int buflen) {
   int ret = child_stream_->read(buf, buflen);
   if (ret > 0) {
-    crc_ = crc32(crc_, buf, ret);
+    crc_ = crc32(gsl::narrow<uLong>(crc_), buf, ret);
```

Comment:
I suggest changing the type of `crc_` to `uint32_t` and `static_assert` that `uLong` has the same width and signedness as `uint32_t`. This should make `gsl::narrow` calls redundant.

Reply:
`uLong` in zlib is a typedef to `unsigned long`, so on 64-bit Linux it is 64 bits long.  I have changed the type of `crc_` to `uLong`, so now we only have one `gsl::narrow` (called once per instance) instead of three (called many times).

Reply:
Oh, I assumed it must be 32 bit, because it's crc32. Great, thanks.